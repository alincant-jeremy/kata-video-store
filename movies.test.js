const statement = require('./movies');

let newReleaseCustomer = {
    "name": "martin",
    "rentals": [
        { "movieID": "F001", "days": 3 },
    ]
};

let newReleaseMovies = {
    "F001": { "title": "The Cell", "code": "new" }
}

test('test single NewRelease statement', () => {
    expect(statement(newReleaseCustomer, newReleaseMovies)).toMatchSnapshot();
});

let newReleaseCustomer2 = {
    "name": "martin",
    "rentals": [
        { "movieID": "F001", "days": 3 },
        { "movieID": "F002", "days": 3 },
    ]
};

let newReleaseMovies2 = {
    "F001": { "title": "The Cell", "code": "new" },
    "F002": { "title": "The Tigger Movie", "code": "new" }
}

test('test dual NewRelease statement', () => {
    expect(statement(newReleaseCustomer2, newReleaseMovies2)).toMatchSnapshot();
});

let newChildrenCustomer = {
    "name": "martin",
    "rentals": [
        { "movieID": "F001", "days": 3 },
    ]
};

let newChildrenMovies = {
    "F001": { "title": "The Tigger Movie", "code": "childrens" }
}

test('test single children statement', () => {
    expect(statement(newChildrenCustomer, newChildrenMovies)).toMatchSnapshot();
});

let regularCustomer = {
    "name": "martin",
    "rentals": [
        { "movieID": "F001", "days": 1 },
        { "movieID": "F002", "days": 2 },
        { "movieID": "F003", "days": 3 },
    ]
};

let regularMovies = {
    "F001": { "title": "Plan 9 from Outer Space", "code": "regular" },
    "F002": { "title": "8 1/2", "code": "regular" },
    "F003": { "title": "Eraserhead", "code": "regular" },
}

test('test multiple regular statement', () => {
    expect(statement(regularCustomer, regularMovies)).toMatchSnapshot();
});
