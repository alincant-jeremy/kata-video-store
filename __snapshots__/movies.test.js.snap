// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dual NewRelease statement 1`] = `
"Rental Record for martin
	The Cell	9
	The Tigger Movie	9
Amount owed is 18
You earned 4 frequent renter points
"
`;

exports[`test multiple regular statement 1`] = `
"Rental Record for martin
	Plan 9 from Outer Space	2
	8 1/2	2
	Eraserhead	3.5
Amount owed is 7.5
You earned 3 frequent renter points
"
`;

exports[`test single NewRelease statement 1`] = `
"Rental Record for martin
	The Cell	9
Amount owed is 9
You earned 2 frequent renter points
"
`;

exports[`test single children statement 1`] = `
"Rental Record for martin
	The Tigger Movie	1.5
Amount owed is 1.5
You earned 1 frequent renter points
"
`;
